integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/testsupport/types/YearMonthFieldTypeDescriptor.java

```diff
@@ -69,7 +69,9 @@ protected List<YearMonth> createSingle() {
 	@Override
 	protected List<YearMonth> createUniquelyMatchableValues() {
 		List<YearMonth> values = new ArrayList<>();
-		for ( Year year : YearFieldTypeDescriptor.INSTANCE.getIndexableValues().getSingle() ) {
+		// NOTE: we are not going with the other years like (createUniquelyMatchableValues) because
+		//  years like 18353 cannot be parsed back from a string for some reason ...
```

Comment:
Years beyond `9999` are supposed to be prefixed with a `+`, e.g. `+10000`; that's the `ISO_YEAR` format used in `ParseUtils`.

I guess `Year.toString()` doesn't output the year in ISO format. If that's the case I'd suggest exposing a `format` method in `FieldTypeDescriptor` and using that instead of `toString()` in relevant tests.

We can also certainly discuss changing the format used for parsing years... But I feel that's a broader change that deserves its own Jira.